### DISCUSSION TITLE
Zキーの連続入力状態

### DISCUSSION THREAD

**Discussion by palchan-net** [Yowkees/keyball#331] [Q&A]
ファームウェアを登録し、各キー入力の動作確認を行おうとしましたが、左手側をUSBケーブル接続時にZキーが押下された状態になります。
他のキーを押すとZキーの入力は止まりますが、その後Zキーを押しても反応しない状態です。
どこかが短絡しているのだと思いますが、特定に至りませんでしたのでご助力いただけますようお願いします。

![image](https://github.com/Yowkees/keyball/assets/143974554/11d12d90-a475-47a6-8a7d-847f620c9d8b)
キースイッチは抜いた状態でも現象は発生します。該当箇所のはんだはつけ直しはしてみましたが変わらずでした。
どう繋がっているのかわからず、お手数ですがご回答の程よろしくお願いします。

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

左手にUSBをつないだ時の不具合ということで、左右の通信に問題があるのかなと推測しました。

下記画像のTRSコネクタの半田付けが弱そうに見えます。
基板の銅パッドまで熱を加えて、半田が銅パッドと馴染み半田が球形ではなく山型になるように修正してみてください。

![KB39_trrsfix](https://github.com/Yowkees/keyball/assets/84209772/bbeb71c8-747c-4378-8df4-7acbcf2300b4)

お手数ですがご確認お願いいたします。

**Reply by palchan-net** [reply]
説明が不十分でした。左右とも単体で検証した際に、左手側で発生した現象となります。

該当箇所は以下を確認して別途着手しておりました。
https://github.com/Yowkees/keyball/discussions/301
（補足：無事左右の接続はできました。あとはこのZが押しっぱなしになる現象が解消すればキーボードとして利用可能になるものと思います）

**Comment by Yowkees** [comment]
左右通信は本件に関係なかったのですね。

下記にKeyball39の回路図と基板データを公開しております。KiCAD等フリーソフトで表示できると思います。
https://github.com/Yowkees/keyball/tree/main/keyball39/design_data

Zキーが常時入力されるということで、COL0線とROW2線がショートしていないかご確認ください。
特にProMicro裏は、この２信号が隣り合って距離が非常に近いので、ここを中心に見てみてください（下記画像の緑点の距離が近いもの同士がショートしていないことをご確認ください）
![KB39_colrowfix](https://github.com/Yowkees/keyball/assets/84209772/335de9a4-c9fb-470f-8bda-476fbc087c91)

また、COL0かROW2がGNDとショートしてしまっていないかもご確認お願いします。

以上、お手数ですがご確認お願いいたします。

**Comment by palchan-net** [comment]
個別でやってもよくわからず、ProMicro裏の半田を全部吸い取ってつけ直したらうまくいきました。
ショートしていた位置は特定できませんでしたが怪しいところは全部やり直すに限るということで、なんとかなりました。ありがとうございました。
また基板データの情報もありがとうございます。次に向けて（keyball61も購入したので）確認させていただこうと思います。
